test-result-summary-client/src/Settings/Settings.jsx

```diff
@@ -81,16 +85,51 @@ export default class Settings extends Component {
         };
     }
 
-    onDelete = async ( key ) => {
+    modalCancel = () => {
+        this.setState({
+            showModal: false,
+        });
+    }
+
+    onDelete = async ( record ) => {
         const { data } = this.state;
-        const target = data.find( item => item.key === key );
+
+        // check if this record already exists
+        if (record._id) {
+            this.setState({ 
+                showModal: true,
+                record: record
+             });
+        } else {
+            // if not, then simply update table without calling API
+            this.setState( { 
+                data: data.filter( item => item.key !== record.key ),
+            } );
+        }
+    }
+
+    deleteBuilds = deleteDataBool => async () => {
+        const { data, record } = this.state;
+
+        // delete build data from database
+        if (deleteDataBool) {
+            await fetch(`/api/deleteBuildData?buildUrl=${record.buildUrl}`, {
+                method: 'get'
+            });
+        }
+
+        // delete the build URL from the Build Monitoring List
+        const target = data.find( item => item.key === record.key );
         if ( target && target._id ) {
             const response = await fetch( `/api/deleteBuildListById?_id=${target._id}`, {
                 method: 'get'
             } );
             await response.json();
         }
-        this.setState( { data: data.filter( item => item.key !== key ) } );
+        this.setState( { 
```

Comment:
We should call deleteBuildListById to stop monitoring, then call deleteBuildData. The reason is that deleteBuildData could take a long time (depends on the amount of history that we have). We do not want TRSS inserting data while we are deleting data at the same time.